mars/services/task/execution/ray/executor.py

```diff
@@ -220,6 +244,10 @@ async def execute_subtask_graph(
         tile_context: TileContext,
         context: Any = None,
     ) -> Dict[Chunk, ExecutionChunkResult]:
+        if self._cancelled is True:  # pragma: no cover
+            raise asyncio.CancelledError()
+        self._execute_subtask_graph_task = asyncio.current_task()
```

Comment:
Whhy do we need to recoed the current aio task? Cae we just call `ray.cancel` when received a `cancel` request?

Reply:
If the subtask is not finished before receiving the `cancel` method, it will still submit more tasks. So we should cancel the subtask firstly.